### ISSUE TITLE
Fleet - fleet-agent fails to deploy in downstream clusters in airgap environment 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#29036]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- run rancher:`v2.5.0-alpha4` docker install in air-gap environment
- add custom clusters 

**Result:**
the `cluster` CRDs for the downstream clusters failes to create.

  
<img width="1625" alt="Screen Shot 2020-09-20 at 8 38 46 AM" src="https://user-images.githubusercontent.com/6218999/93715257-1ef1ae00-fb1d-11ea-89b3-fbe545072731.png">
<img width="1631" alt="Screen Shot 2020-09-20 at 8 38 58 AM" src="https://user-images.githubusercontent.com/6218999/93715261-21ec9e80-fb1d-11ea-8304-718f972d785e.png">

**Comment by deniseschannon** [comment]
@jiaqiluo @StrongMonkey Is this cause we haven't set up to handle the system default registry? or is it because of https://github.com/rancher/rancher/issues/29026

**Comment by StrongMonkey** [comment]
fleet should already support airgap starting from v0.3.0-beta2. I will take a look if it is a broader issue.

**Comment by maggieliu** [comment]
@jiaqiluo Can you check if airgap works if we apply the workaround?

**Comment by jiaqiluo** [comment]
in a custom cluster in rancher: `v2.5.0-alpha4` docker install in air-gap environment

restarting the `cattle-cluster-agent` pods does NOT sovle the problem

cc @maggieliu @StrongMonkey

**Comment by StrongMonkey** [comment]
@jiaqiluo Can you also test this on a HA airgap setup?

TLDR; fleet doesn't work in docker install airgap, this is because fleet-controller can't reach the rancher server URL(thus downstream cluster). This is because for QA docker install airgap, server URL is set to the private dns name that points to private IP that the pod(fleet-controller) in local cluster that lives in docker install failed to reach.

**Comment by StrongMonkey** [comment]
After talking to darren, we won't be able to fix this in 2.5. 
```
At this point, fleet-controller has to be able to talk to rancher server URL in order to function properly.
```

Post 2.5 we might have to introduce a hack to address the docker install issue if needed. If we are getting rid of `docker install`, then fleet-controller can use `rancher.cattle-system` dns name to take to rancher pods using service IP, which is a more efficient  way.

**Comment by sangeethah** [comment]
Blocked by https://github.com/rancher/rancher/issues/29143

**Comment by jiaqiluo** [comment]
This bug fix is validated on rancher:v2.5.0-rc2 HA airgap setup 

the fleet-agent is deployed successfully in the downstream custom cluster.
